docs/advanced/customization-guide.md

```diff
@@ -16,8 +15,1007 @@ published: false
+`/etc/kubernetes/node-feature-discovery/features.d/` in the side-car (e.g.
+device plugin) creates a shared area for deploying hooks and feature files to
+NFD. NFD will periodically scan the directories and run any hooks and read any
+feature files it finds.
+
+## Custom feature source
+
+The *custom* feature source in nfd-worker provides a rule-based mechanism for
+label creation, similar to the
+[NodeFeatureRule](#nodefeaturerule-custom-resource) objects. The difference is
+that the rules are specified in the worker configuration instead of a
+Kubernetes API object.
+
+See [worker configuration](deployment-and-usage.md#worker-configuration) for
+instructions how to set-up and manage the worker configuration.
+
+### An example custom feature source configuration
+
+Consider the following referential configuration for nfd-worker:
+
+```yaml
+core:
+  labelSources: ["custom"]
+sources:
+  custom:
+    - name: "my sgx feature rule"
+      labels:
+        "my-sgx-feature": "true"
+      matchFeatures:
+        - feature: cpu.cpuid
+          matchExpressions:
+            SGX: {op: Exists}
+        - feature: kernel.config
+          matchExpressions:
+            X86_SGX: {op: In, value: ["y"]}
+```
+
+It specifies one rule which creates node label
+`feature.node.kubenernetes.io/my-sgx-feature=true` if both of the following
+conditions are true (`matchFeatures` implements a logical AND over the
+matchers):
+
+- the CPU has SGX capability
+- X86_SGX option has been enabled in the kernel
+
+In addition, the configuration only enables the *custom* source, disabling all
+built-in labels.
+
+### Additional configuration directory
+
+Additionally to the rules defined in the nfd-worker configuration file, the
+Custom feature can read more configuration files located in the
+`/etc/kubernetes/node-feature-discovery/custom.d/` directory. This makes more
+dynamic and flexible configuration easier.
+
+As an example, consider having file
+`/etc/kubernetes/node-feature-discovery/custom.d/my-rule.yaml` with the
+following content:
+
+```yaml
+- name: "my e1000 rule"
+  labels:
+    "e1000.present": "true"
+  matchFeatures:
+    - feature: kernel.loadedmodule
+      matchExpressions:
+        e1000: {op: Exists}
+```
+
+This simple rule will create `feature.node.kubenernetes.io/e1000.present=true`
+label if the `e1000` kernel module has been loaded.
+
+The
+[`samples/custom-rules`](https://github.com/kubernetes-sigs/node-feature-discovery/blob/{{site.release}}/deployment/overlays/samples/custom-rules)
+kustomize overlay sample contains an example for deploying a custom rule from a
+ConfigMap.
+
+## Node labels
+
+Feature labels have the following format:
+
+```plaintext
+<namespace>/<name> = <value>
+```
+
+The namespace part (i.e. prefix) of the labels is controlled by nfd:
+
+- All built-in labels use `feature.node.kubernetes.io`. This is also
+  the default for user defined features that don't specify any namespace.
```

Comment:
and that `<name>` includes the source as prefix?

Reply:
For built-in labels, yes. But I don't think it needs to be stated here(?)

Reply:
Did not change this